mitm/mitm.go

```diff
@@ -51,6 +52,7 @@ type Config struct {
 	getCertificate func(*tls.ClientHelloInfo) (*tls.Certificate, error)
 	roots          *x509.CertPool
 	skipVerify     bool
+	TLSHandshakeErrorHandler func(*http.Request, error)
```

Comment:
Maybe "TLSHandshakeErrorCallback"? "Handler" implies the error is being dealt with/gotten rid of, rather than just being a notification.